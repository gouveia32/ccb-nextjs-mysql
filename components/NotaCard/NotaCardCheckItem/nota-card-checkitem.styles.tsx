import styled from "styled-components";

export const NotaCardCheckItemComponent = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
`;

export const NotaCardCheckItemCheck = styled.div`
  //margin-right: 1rem;
`;

interface NotaCardCheckItemTextInterface {
  checked: boolean;
}

export const NotaCardCheckItemText = styled.span<
  NotaCardCheckItemTextInterface
>`
  flex: 1;
  display: flex;
  align-items: center;
  text-decoration: ${(props) => (props.checked ? "line-through" : "none")};
`;
