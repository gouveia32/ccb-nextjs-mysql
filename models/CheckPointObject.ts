export interface CheckPointInterface {
    id?: number | string;
    text: string;
    checked: boolean;
  }
  
  export type CheckPointType = CheckPointInterface;
  
  export const CheckPointObject: CheckPointType = {
           id: 0,
           text: "",
           checked: false,
         };
  
  export enum cCheckPoint {
    id = 'id',
    text = 'text',
    checked = 'checked',
  }