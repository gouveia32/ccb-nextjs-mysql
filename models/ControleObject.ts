export interface ControleInterface {
  id?: number | string;
  texto: string;
  marcado: boolean;
}

export type TipoControle = ControleInterface;

export const ControleObject: TipoControle = {
         id: 0,
         texto: "",
         marcado: false,
       };

export enum cControle {
  id = 'id',
  texto = 'texto',
  marcado = 'marcado',
}
