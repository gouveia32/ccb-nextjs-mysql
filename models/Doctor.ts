export interface DoctorModel {
  id: string;
  name: string;
  email: string;
  image?: string;
}

export type DoctorType = DoctorModel;

export const cDoctorModel = {
  id: "id",
  name: "name",
  email: "email",
  image: '',
};

export const DoctorObject: DoctorType = {
  id: "",
  name: "",
  email: "",
  image: '',
};

export function isDoctorType(object: any): object is DoctorType {
  return "name" in object;
}
