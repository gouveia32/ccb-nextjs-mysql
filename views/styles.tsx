import styled from "styled-components";
import { motion } from "framer-motion";
import { device } from "../resources/styles/utils/media-query-utils";

const dFlexCentered = `display: flex;
  justify-content: center;
  align-items: center;`;

export const LandingPageWrapper = styled(motion.section)`
  height: 100%;
  flex-direction: column;
  ${dFlexCentered}

  @media only screen and ${device.mobileL} {
    height: max-content;
    margin-left: -50px;
    margin-top: 2rem;
  }
`;

export const LandingPageContentWrapper = styled(motion.div)`
  ${dFlexCentered}
  
  @media only screen and ${device.mobileL} {
    flex-direction: column;
  }
`;

export const LandingPageCardWrapper = styled(motion.div)`
  flex-direction: column;
  ${dFlexCentered}

  @media only screen and ${device.mobileL} {
    margin: 0 !important;
  }
`;

export const LandingPageHeading = styled.h1`
  margin-bottom: 1rem;
  font-weight: bold;
  font-size: 4rem;
  color: #f5b500;
`;

export const LandingPageSubHeading = styled.h2`
  margin-top: 2rem;
  margin-bottom: 1rem;
  color: #8a8a8a;
`;

interface LandingPageBannerInterface {
  imageUrl?: string;
  color?: string;
  position?: string;
}

export const LandingPageBannerWrapper = styled.div`
  position: relative;
  height: 300px;
  width: 300px;
  overflow: hidden;
`;

export const LandingPageBanner = styled(motion.div)<LandingPageBannerInterface>`
  height: 300px;
  width: 300px;
  transform-origin: top;
  font-size: 2.8rem;
  font-family: "Roboto Slab", serif;
  font-weight: lighter;
  padding: 1rem;
  top: 0;
  background-size: cover;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  position: ${(props) => (props.position ? props.position : "relative")};
  background: ${(props) =>
    props.imageUrl
      ? `url(${props.imageUrl})`
      : props.color
      ? props.color
      : "#ffe900"};
`;

export const LandingPageBannerText = styled(motion.h1)`
  font-family: "Shadows Into Light Two", cursive;
  font-size: 1.8rem;
  left: 0;
  padding: 10px;
  position: absolute;
  right: 0;
  text-align: center;
  top: 50%;
  -webkit-transform: translateY(-50%) rotate(-4deg);
  transform: translateY(-50%) rotate(-4deg);
`;
