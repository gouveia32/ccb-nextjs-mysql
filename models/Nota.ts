import { TipoPaciente } from "./Paciente";
import { TipoControle } from "./ControleObject";
import { Nota } from "@prisma/client";

export enum tipoNotaEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NotaModel {
  id?: string;
  nome: string;
  tipoNota: tipoNotaEnum;
  conteudo: string;
  cor: string;
  pino: boolean;
  criadoEm: string;
  pacientes: TipoPaciente[];
  controles?: TipoControle[];
}

export type TipoNota = NotaModel;

export const NotaObject: TipoNota = {
  id: "",
  nome: "",
  tipoNota: tipoNotaEnum.TEXT,
  conteudo: "",
  cor: "#fff",
  pino: false,
  criadoEm: new Date(Date.now()).toString(),
  pacientes: [],
  controles: [],
};

export const NotaDBObject: Nota = {
  id: "",
  nome: "",
  tipoNota: "TEXT",
  cor: "",
  conteudo: "",
  pino: false,
  criadoEm: new Date(Date.now()),
  medicoId: "",
};

export enum cNotaModel {
  id = "id",
  nome = "nome",
  tipoNota = "tipoNota",
  cor = "cor",
  conteudo = "conteudo",
  pino = "pino",
  criadoEm = "criadoEm",
  pacientes = "pacientes",
  controles = "controles",
}
