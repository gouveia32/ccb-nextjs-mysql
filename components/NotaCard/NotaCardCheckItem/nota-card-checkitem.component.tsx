import React from "react";
import {cCheckPoint, CheckPointType} from "../../../models/ControleObject";
import {Checkbox} from "@material-ui/core";
import {NotaCardCheckItemCheck, NotaCardCheckItemComponent, NotaCardCheckItemText,} from "./nota-card-checkitem.styles";

export interface NotaCardCheckItemProps {
  checkItem: CheckPointType;
  onChecked?: (checkItem: CheckPointType) => void;
}

const NotaCardCheckItem: React.FC<NotaCardCheckItemProps> = ({
  checkItem,
  onChecked,
}: NotaCardCheckItemProps) => {
  const handleChange = (attr: string, event: any) => {
    const newCheckItem: any = { ...checkItem };

    newCheckItem[attr] = event.target.checked;

    onChecked && onChecked(newCheckItem);
  };

  return (
    <NotaCardCheckItemComponent>
      <NotaCardCheckItemCheck>
        <Checkbox
          size={"small"}
          color={"default"}
          checked={checkItem.checked}
          className="p-1"
          onChange={(event) => handleChange(cCheckPoint.checked, event)}
          onClick={(event) => event.stopPropagation()}
        />
      </NotaCardCheckItemCheck>
      <NotaCardCheckItemText checked={checkItem.checked}>
        {checkItem.text}
      </NotaCardCheckItemText>
    </NotaCardCheckItemComponent>
  );
};

export default NotaCardCheckItem;
