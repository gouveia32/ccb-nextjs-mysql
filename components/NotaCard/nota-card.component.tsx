import React, { useState } from "react";
import { NotaObject, TipoNota, tipoNotaEnum } from "../../models/Nota";
import { TipoControle } from "../../models/ControleObject";
import NotaCardCheckItem from "./NotaCardCheckItem/nota-card-checkitem.component";
import { TipoPaciente } from "../../models/Paciente";
import { ChangeActionType } from "../../lib/helpers";
import { Dialog, Divider, IconButton } from "@material-ui/core";
import {
  NotaCardComponent,
  NotaCardConteudo,
  NotaCardHeader,
  NotaCardPaciente,
  NotaCardPacientes,
} from "./nota-card.styles";
import DeleteOutlineOutlinedIcon from "@material-ui/icons/DeleteOutlineOutlined";
import LabelOutlinedIcon from "@material-ui/icons/LabelOutlined";
import AddNota from "../AddNota/add-nota.component";

const transition = {
  type: "spring",
  stiffness: 100,
};

const notaVariants = {
  exit: { y: "50%", opacity: 0, transition },
  enter: {
    y: "0%",
    opacity: 1,
    transition,
  },
};

export interface NotaCardProps {
  nota: TipoNota;
  pacientes: TipoPaciente[];
  editNota: TipoNota | null;
  onHandleChange: (action: ChangeActionType) => void;
  onAddNota: () => void;
  onDeleteNota: () => void;
  onClick?: () => void;
  onCheckItemClick?: (checkItem: TipoControle) => void;
  onCloseModal?: () => void;
}

const NotaCard: React.FC<NotaCardProps> = ({
  nota,
  pacientes,
  editNota,
  onAddNota,
  onHandleChange,
  onCloseModal,
  onClick,
  onCheckItemClick,
  onDeleteNota,
}: NotaCardProps) => {
  const [modalOpen, setModalOpen] = useState(false);

  const handleOnDelete = (e: any) => {
    e.stopPropagation();
    onDeleteNota();
  };

  const renderHeader = (
    <NotaCardHeader>
      <span>{nota.nome}</span>
      <IconButton onClick={handleOnDelete} size={"small"}>
        <DeleteOutlineOutlinedIcon />
      </IconButton>
    </NotaCardHeader>
  );

  const renderContent =
    nota.tipoNota === tipoNotaEnum.TEXT ? (
      <NotaCardConteudo>{nota.conteudo}</NotaCardConteudo>
    ) : (
      <NotaCardConteudo>
        {nota.controles
          ?.filter((f) => !f.marcado)
          .map((cp: TipoControle, i: number) => (
            <NotaCardCheckItem
              key={i}
              checkItem={cp}
              onChecked={onCheckItemClick}
            />
          ))}
        {nota.controles?.find((f) => f.marcado) && (
          <Divider className="bg-dark" />
        )}
        {nota.controles
          ?.filter((f) => f.marcado)
          .map((cp: TipoControle, i: number) => (
            <NotaCardCheckItem
              key={i}
              checkItem={cp}
              onChecked={onCheckItemClick}
            />
          ))}
      </NotaCardConteudo>
    );

  const renderPacientes = nota.pacientes.length > 0 && (
    <>
      <Divider className="my-2 bg-dark" />
      <NotaCardPacientes>
        {nota.pacientes.map((paciente: TipoPaciente, k: number) => (
          <NotaCardPaciente key={k}>
            <LabelOutlinedIcon className="me-1" />
            <span>{paciente.nome}</span>
          </NotaCardPaciente>
        ))}
      </NotaCardPacientes>
    </>
  );

  const renderEditModal = (
    <Dialog
      fullWidth={true}
      open={modalOpen && !!editNota}
      classes={{
        paper: "bg-transparent",
      }}
      onClose={() => {
        setModalOpen((prevState) => !prevState);
        onCloseModal && onCloseModal();
      }}
    >
      <AddNota
        onHandleChange={onHandleChange}
        notaModel={editNota ? editNota : NotaObject}
        pacientes={pacientes}
        onAddNota={onAddNota}
        edit={true}
      />
    </Dialog>
  );

  return (
    <>
      <NotaCardComponent
        onClick={() => {
          setModalOpen(!modalOpen);
          if (onClick) {
            onClick();
          }
        }}
        color={nota.cor}
        variants={notaVariants}
        initial="exit"
        animate="enter"
        exit="exit"
      >
        {renderHeader}
        {renderContent}
        {renderPacientes}
      </NotaCardComponent>
      {renderEditModal}
    </>
  );
};

export default React.memo(NotaCard);
