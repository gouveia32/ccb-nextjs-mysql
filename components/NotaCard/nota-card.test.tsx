import * as React from "react";
import {shallow, ShallowWrapper} from "enzyme";
import {Dialog, Divider, IconButton} from "@material-ui/core";
import NotaCard, {NotaCardProps} from "./nota-card.component";
import {NotaObject, NotaTypeEnum} from "../../models/Nota";
import {NotaCardComponent} from "./nota-card.styles";
import {CheckPointObject} from "../../models/ControleObject";
import NotaCardCheckItem from "./NotaCardCheckItem/nota-card-checkitem.component";

describe("NotaCard component", () => {
  let wrapper: ShallowWrapper;
  let mockOnHandleChange: any;
  let mockOnAddNota: any;
  let mockOnDeleteNota: any;
  let mockOnClick: any;
  let mockOnCheckItemClick: any;
  let mockOnCloseModal: any;

  beforeEach(() => {
    mockOnHandleChange = jest.fn();
    mockOnAddNota = jest.fn();
    mockOnDeleteNota = jest.fn();
    mockOnClick = jest.fn();
    mockOnCheckItemClick = jest.fn();
    mockOnCloseModal = jest.fn();

    const mockProps: NotaCardProps = {
      nota: NotaObject,
      tags: [],
      editNota: NotaObject,
      onHandleChange: mockOnHandleChange,
      onAddNota: mockOnAddNota,
      onDeleteNota: mockOnDeleteNota,
      onClick: mockOnClick,
      onCheckItemClick: mockOnCheckItemClick,
      onCloseModal: mockOnCloseModal,
    };

    wrapper = shallow(<NotaCard {...mockProps} />);
  });

  it("should render NotaCard component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onClick when NotaCardComponent component is clicked", () => {
    wrapper.find(NotaCardComponent).simulate("click");
    expect(mockOnClick).toHaveBeenCalled();
  });

  it("should call onClose when Dialog component closed", () => {
    wrapper.find(Dialog).simulate("close");
    expect(mockOnCloseModal).toHaveBeenCalled();
  });

  it("should call onDelete when IconButton is clicked", () => {
    wrapper
      .find(IconButton)
      .simulate("click", { stopPropagation: () => undefined });
    expect(mockOnDeleteNota).toHaveBeenCalled();
  });

    it("should render NotaCardCheckItem and Divider components when notaType is CHECK", () => {
        const notaModel = NotaObject;
        notaModel.notaType = NotaTypeEnum.CHECK;

        const checkedCheckpoint = CheckPointObject;
        checkedCheckpoint.checked = true;
        notaModel.checkPoints = [CheckPointObject, checkedCheckpoint];

        const mockProps: NotaCardProps = {
            nota: NotaObject,
            tags: [],
            editNota: NotaObject,
            onHandleChange: mockOnHandleChange,
            onAddNota: mockOnAddNota,
            onDeleteNota: mockOnDeleteNota,
            onClick: mockOnClick,
            onCheckItemClick: mockOnCheckItemClick,
            onCloseModal: mockOnCloseModal,
        };

        wrapper = shallow(<NotaCard {...mockProps} />);

        const notaCardCheckpoints = wrapper.find(NotaCardCheckItem);
        expect(notaCardCheckpoints).toHaveLength(2);

        const divider = wrapper.find(Divider);
        expect(divider).toHaveLength(1);
    });
});
