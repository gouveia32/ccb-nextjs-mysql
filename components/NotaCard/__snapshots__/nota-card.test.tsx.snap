// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotaCard component should render NotaCard component 1`] = `ShallowWrapper {}`;
